DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift

```diff
@@ -137,38 +158,70 @@ extension YoutubePlayerNavigationHandler: DuckNavigationHandling {
         if let url = navigationAction.request.url,
             let (videoID, timestamp) = url.youtubeVideoParams,
             duckPlayer.settings.mode == .disabled {
-            webView.load(URLRequest(url: URL.youtube(videoID, timestamp: timestamp)))
-            completion(.allow)
+            os_log("DP: is Disabled. We should load original video for %s", log: .duckPlayerLog, type: .debug)
+            handleURLChange(url: URL.youtube(videoID, timestamp: timestamp), webView: webView)
             return
         }
-        
-        completion(.allow)
-        
     }
     
     // Handle URL changes not triggered via Omnibar
     // such as changes triggered via JS
     @MainActor
     func handleURLChange(url: URL?, webView: WKWebView) {
         
+        // Do not handle the URL if the video was just handled
+        if let url = url,
+           url.isYoutubeVideo || url.isDuckPlayer,
+           let (videoID, timestamp) = url.youtubeVideoParams,
+            lastHandledVideoID == videoID,
+            !isDuckPlayerTemporarilyDisabled {
+                return
+        }
+        
         if let url = url, url.isYoutubeVideo,
             !url.isDuckPlayer,
             let (videoID, timestamp) = url.youtubeVideoParams,
-           duckPlayer.settings.mode == .enabled || duckPlayer.settings.mode == .alwaysAsk {
-            webView.stopLoading()
-            let newURL = URL.duckPlayer(videoID, timestamp: timestamp)
+            duckPlayer.settings.mode == .enabled || duckPlayer.settings.mode == .alwaysAsk {
+            
+            os_log("DP: Handling URL change: %s", log: .duckPlayerLog, type: .debug, url.absoluteString)
+            var newURL = URL.duckPlayer(videoID, timestamp: timestamp)
+            
+            // IF DP is temporarily disabled, load Youtube website
+            // Then reset the setting
+            if isDuckPlayerTemporarilyDisabled {
+                os_log("DP: Duckplayer is temporarily disabled.  Opening Youtube", log: .duckPlayerLog, type: .debug)
+                newURL = URL.youtube(videoID, timestamp: timestamp)
+            } else {
+                os_log("DP: Duckplayer is NOT disabled.  Opening DuckPlayer", log: .duckPlayerLog, type: .debug)
+            }
+            
+            // Load the URL
             webView.load(URLRequest(url: newURL))
+            
+            // Add a delay before resetting to allow the Webview to properly render
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Isn't there a webview callback to listen instead of adding delays?

Reply:
This is more about letting the webView start the loading process, not rendering the page.  The only callback is the WKNavigation delegate, but we don't want anything to do with that here.   I'll update the comment so is more clear :)